src/include/mallocMC/reservePoolPolicies/CudaSetLimits_impl.hpp

```diff
@@ -27,26 +27,40 @@
 
 #pragma once
 
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+
 #include "CudaSetLimits.hpp"
 
 #include <cuda_runtime_api.h>
+#include <mutex>
 #include <string>
 
 namespace mallocMC
 {
     namespace ReservePoolPolicies
     {
+        // TODO alpaka
         struct CudaSetLimits
         {
-            static auto setMemPool(size_t memsize) -> void *
+            template<typename AlpakaDev>
+            auto setMemPool(const AlpakaDev & dev, size_t memsize) -> void *
             {
-                cudaDeviceSetLimit(cudaLimitMallocHeapSize, memsize);
+                // FIXME: cannot call cudaDeviceSetLimit() anymore if any kernel has been launched
+                static auto first = cudaDeviceSetLimit(
+                    cudaLimitMallocHeapSize,
+                    memsize); // abusing thread-safe static init for calling
+                              // cudaDeviceSetLimit only once per process
```

Comment:
According to CUDA documentation I cannot call this function anymore if a kernel has been run, that used malloc/free. This happens however during the unit tests if you run 2 unit tests, where each one creates an allocator with the `CudaSetLimits` policy.